src/utils.cpp

```diff
@@ -3742,6 +3742,47 @@ int get_string_length_after_color_code_removal(const char *str, struct char_data
   return len;
 }
 
+// Returns a string stripped of color for keyword matching or possibly other uses as well.
+// We don't need to check for color code validity because we call get_string_legth_after_color_code_removal() when the strings are initially written.
+char* get_string_after_color_code_removal(const char *str, struct char_data *ch_to_notify_of_failure_reason) {
+  if (!str) {
+    mudlog("SYSERR: Null string received to get_string_after_color_code_removal().", ch_to_notify_of_failure_reason, LOG_SYSLOG, TRUE);
```

Comment:
This doesn't actually notify the character, it just writes a syslog that has their name in it. You'll want to either send_to_char() them a message, or have the code that calls this send the error message.

Reply:
Hi!

- The string is already nulll terminated because we copy char after char from the original null terminated string hence '\0' is being copied at the end as well. The function is just a copier, if the original string is null terminated then the resulting string will also be null terminated. I have already checked for this and the strings used in obj->restring are null terminated and my resulting return strings were also null terminated.
I could check at the end of the copy loop if `clearstr[pos] != '\0' and insert null termination on `clearstr[pos+1]` if not at max length or overwrite `clearstr[pos]` with null termination if it is (it's cheaper in cycles than zeroing clearstr). But I feel that this would just limit the function's usability (won't be able to return non-null terminated strings if that was required) and it's really the wrong place to ensure null termination as that should happen when writing the original strings to begin with (and it happens there already). This is also why I'm not doing any color code format testing as well because it is a waste of cycles when we have already done that in `get_string_length_after_color_code_removal()` when length checking, prior to writing the original strings.
Let me know if you want me to do this and limit the functionality to null terminated strings only. If we keep it as is, I can add a comment that the function will return the same type of string as what is passed to it (null terminated or not) for clarity.

- I didn't do this because the function is expecting proper existing strings up to `MAX_STRING_LENGTH` to convert to plaintext. I added the normal string length checks to the functions that were writing the original strings to begin with using `get_string_length_after_color_code_removal()` like it's done everywhere else. But I could add this for redundancy/fail safe if you want me to.

- Regading the mudlog() call, I know it doesn't send anything to the character heh. All errors that the character should be aware of are already handled in the caller functions and in `get_string_length_after_color_code_removal()`. I just added this syslog entry for consistency because `get_string_length_after_color_code_removal()` also did the same. I have a habit of conforming to debugging practices that are already present in the code :)

Let me know what you want me to do about 1 and 2!